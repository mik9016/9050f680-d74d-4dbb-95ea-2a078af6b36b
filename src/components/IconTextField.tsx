import { TextField, InputAdornment } from '@mui/material'
import React, { useState, useCallback, useContext, useEffect } from 'react'
import { GlobalContext } from '../contexts/Context'

interface Props {
  iconStart: React.ReactElement
  label: string
}

export const IconTextField = ({
  iconStart,
  label,
}: Props): React.ReactElement => {
  const globalCtx = useContext(GlobalContext)

  const [searchValue, setSearchValue] = useState('')

  useEffect(() => {
    globalCtx.setCurrentSearch(searchValue)
  }, [searchValue])

  const changeHandler = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>): void => {
      setSearchValue(e.currentTarget.value)
    },
    []
  )

  return (
    <TextField
      label={label}
      InputProps={{
        startAdornment: (
          <InputAdornment position="start">{iconStart}</InputAdornment>
        ),
      }}
      variant="outlined"
      onChange={changeHandler}
    />
  )
}
