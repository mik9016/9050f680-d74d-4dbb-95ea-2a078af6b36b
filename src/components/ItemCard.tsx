import {
  Card,
  CardActions,
  CardContent,
  CardMedia,
  IconButton,
  CardHeader,
  Button,
  Avatar,
  Typography,
} from '@mui/material'
import AddCircleIcon from '@mui/icons-material/AddCircle'
import React, { useContext } from 'react'
import LocationOnIcon from '@mui/icons-material/LocationOn'
import { GlobalContext } from '../contexts/Context'

export interface Props {
  bandName: string
  avatar: string
  picture: string
  venue: string
  venueName: string
  start: string
  end: string
}

const ItemCard = ({
  bandName,
  avatar,
  picture,
  venue,
  start,
  venueName,
  end,
  ref,
}: Props & { ref?: React.Ref<HTMLDivElement> }): React.ReactElement => {
  const globalCtx = useContext(GlobalContext)

  const infos = {
    bandName: bandName,
    avatar: avatar,
    picture: picture,
    venue: venue,
    start: start,
    venueName: venueName,
    end: end,
  }
  const incrementCartNumber = () => {
    globalCtx.incrementNumber(infos)
  }
  const startDate = new Date(start)
  const endDate = new Date(end)

  const noPicUrl = 'https://picsum.photos/200/300'

  return (
    <Card
      ref={ref}
      sx={{
        width: { xs: 310, md: 345 },
        height: 400,
        margin: { xs: 0, md: 2 },
        my: { xs: 1 },
      }}
    >
      <CardHeader
        avatar={<Avatar alt="Avatar" src={avatar} />}
        title={bandName}
      />
      <CardMedia
        component="img"
        height="140"
        image={picture ? picture : noPicUrl}
        alt="picture"
      />
      <CardContent>
        <Typography gutterBottom variant="h5" component="div">
          <Button target="_blank" href={venue} rel="noopener">
            {' '}
            <LocationOnIcon color="primary" />
            {venueName}{' '}
          </Button>
        </Typography>
        <Typography
          gutterBottom
          variant="body2"
          color="text.secondary"
          component="div"
        >
          {`| Starts: ${startDate.getDate()}.${
            startDate.getMonth() + 1
          }.${startDate.getFullYear()},  ${startDate.getHours()}:${
            startDate.getMinutes() === 0 ? '00' : startDate.getMinutes()
          }`}
        </Typography>
        <Typography variant="body2" color="text.secondary">
          {`| Ends: ${endDate.getDate()}.${
            endDate.getMonth() + 1
          }.${endDate.getFullYear()},  ${endDate.getHours()}:${
            endDate.getMinutes() === 0 ? '00' : endDate.getMinutes()
          }`}
        </Typography>
      </CardContent>
      <CardActions>
        <IconButton aria-label="add" onClick={incrementCartNumber}>
          <AddCircleIcon color="primary" />
        </IconButton>
      </CardActions>
    </Card>
  )
}

export default ItemCard
