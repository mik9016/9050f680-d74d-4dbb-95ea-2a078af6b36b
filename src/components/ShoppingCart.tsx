import { Badge } from '@mui/material'
import React from 'react'
import ShoppingCartOutlinedIcon from '@mui/icons-material/ShoppingCartOutlined'

interface Props {
  number: number
}

export const ShoppingCart = ({ number }: Props): React.ReactElement => {
  return (
    <Badge badgeContent={number} color="error">
      <ShoppingCartOutlinedIcon color="action" />
    </Badge>
  )
}
