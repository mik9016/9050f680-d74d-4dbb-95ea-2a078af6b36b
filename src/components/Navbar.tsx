import React, { FC, ReactElement, useState, useContext } from 'react'
import { Link } from 'react-router-dom'
import { Box, Container, Toolbar } from '@mui/material'
import SearchIcon from '@mui/icons-material/Search'
import { GlobalContext } from '../contexts/Context'

import { IconTextField } from './IconTextField'
import { ShoppingCart } from './ShoppingCart'

const Navbar = (): React.ReactElement => {
  const globalCtx = useContext(GlobalContext)

  return (
    <Box
      sx={{
        width: '100%',
        height: 'auto',
        padding: 1,
        paddingLeft: 3,
        backgroundColor: 'primary.main',
        position: 'fixed',
        zIndex: '4',
      }}
    >
      <Container maxWidth="xl">
        <Toolbar disableGutters>
          <IconTextField label="Search" iconStart={<SearchIcon />} />
          <Box sx={{ flexGrow: 1, display: 'flex' }}>
            <Box
              sx={{
                display: 'flex',
                flexDirection: 'row',
                justifyContent: 'flex-end',
                alignItems: 'center',
                marginLeft: 'auto',
                paddingLeft: '1rem',
              }}
            >
              <Link to={'/cart'}>
                <ShoppingCart number={globalCtx.currentCartNumber} />
              </Link>
            </Box>
          </Box>
        </Toolbar>
      </Container>
    </Box>
  )
}

export default Navbar
