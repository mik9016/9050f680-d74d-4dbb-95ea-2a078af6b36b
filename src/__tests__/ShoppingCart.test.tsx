import React from 'react'

import { act, create } from 'react-test-renderer'

import { ShoppingCart } from '../components/ShoppingCart'

test(`renders ShoppingCart without crashing`, async () => {
  const root = create(<ShoppingCart number={5} />)

  expect(root.toJSON()).toMatchSnapshot()

  act(() => {
    root.unmount()
  })
})
