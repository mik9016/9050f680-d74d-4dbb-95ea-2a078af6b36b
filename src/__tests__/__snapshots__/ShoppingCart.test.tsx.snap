// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ShoppingCart without crashing 1`] = `
<span
  className="MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeMedium css-11pbyhm-MuiSvgIcon-root"
    data-testid="ShoppingCartOutlinedIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.37-.66-.11-1.48-.87-1.48H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
    />
  </svg>
  <span
    className="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError BaseBadge-badge css-to4x1n-MuiBadge-badge"
  >
    5
  </span>
</span>
`;
